### ISSUE TITLE
"Open Empty Editor On Startup" option not working.

### ISSUE THREAD

**Issue by ghost** [atom/atom#10623]
Option "Open Empty Editor On Startup" is not working.
image: http://i.stack.imgur.com/zJrRp.png

It doesn't matter if i tick or untick it, it always opens my last opened project and opened files.

Doesn't work on windows (win10) and linux (Mint 17 Cinnamon).
Tried to disable all plugins. Still having this issue.

If i can give You any information, log file, etc, let me know.

**Comment by lee-dohm** [comment]
Related to: https://github.com/atom/atom/issues/10517

**Comment by brentshermana** [comment]
Confirmed on Ubuntu 14.04.3 with Atom 1.6.0-dev-f914d18

To be clear, is it just the tabs which should not be brought up, or that and all files in the project tree on the left hand side?

**Comment by ExClouds** [comment]
It always opens both:
* project folder on the left side
* previously opened files (doesn't matter if the files were project or manually opened files) (even atom settings pages, like init, keymaps, settings, etc)

**Comment by brentshermana** [comment]
I understand that, but what is it *meant* to do?

**Comment by ExClouds** [comment]
Well, the checkbox say's **Open Empty Editor On Startup**, so, when it's ticked, i expect the editor to open blank.. No open tabs, no open projects..

**Comment by demhydraz** [comment]
Can confirm on Arch Linux with 1.6.0-beta3.

**Comment by minsis** [comment]
Confirmed on Ubuntu Mint 15.10 and Windows 7 Ent.

**Comment by zzeroo** [comment]
Also here on debian sid with 1.6.0-beta3

**Comment by demhydraz** [comment]
Can still confirm on beta6.

**Comment by sp82** [comment]
Same on Windows7

**Comment by Kingdaro** [comment]
And on Windows 10. Sorry to add more confirmations than solutions, but at least we're a _little_ closer to knowing that this is totally an issue on every platform!

**Comment by demhydraz** [comment]
Still happening on beta7, same platform.

**Comment by delliott2016** [comment]
The option causes a different problem on Mac OS X (v10.11.3) using Atom 1.8.0-dev-3959259. When the box is checked, the workspace opens up empty with a single untitled tab. When the box is unchecked, the workspace opens up completely empty.

**Comment by leobasilio** [comment]
That's the actual behavior (given the current code) when no state is saved/restored. Are you working with projects and/or leaving opened tabs when closing atom? If you are, they should be restored on next launch. If they don't, maybe there's something wrong with states themselves on OS X.

**Comment by delliott2016** [comment]
Allow me to add some more information to my description. When only one window is open then the behavior is as I described in my first comment. When there is more than one window open, however, the behavior becomes the same as Windows/Linux. I'll add replication instructions for all of it just for reference. Note that this is only on Mac OS X.

#### One Window
###### Checked

1. Clear out Atom by opening it, closing all tabs and removing all projects in all windows, and quitting the program.
2. Open Atom. You should see either nothing, or a single untitled tab.
3. Open settings through Atom -> Preferences -> Settings and make sure that the "Open Empty Editor On Start" button under Core Settings is selected.
4. Add a project using File -> Add Project Folder and open a file from that project.
5. Close the program by either clicking the red "x" in the corner of the window, or by right-clicking the icon in the menu bar and selecting "Quit".
6. Open Atom again.
7. There is now only a single untitled tab.

###### Unchecked

1. Clear out Atom by opening it, closing all tabs and removing all projects in all windows, and quitting the program.
2. Open Atom. You should see either nothing, or a single untitled tab.
3. Open settings through Atom -> Preferences -> Settings and make sure that the "Open Empty Editor On Start" button under Core Settings is *not* selected.
4. Add a project using File -> Add Project Folder and open a file from that project.
5. Close the program by either clicking the red "x" in the corner of the window, or by right-clicking the icon in the menu bar and selecting "Quit".
6. Open Atom again.
7. Atom opens completely empty, without even an untitled tab present.

#### Multiple Windows
###### Checked and Unchecked

1. Clear out Atom by opening it, closing all tabs and removing all projects in all windows, and quitting the program.
2. Open Atom. You should see either nothing, or a single untitled tab.
3. Open settings through Atom -> Preferences -> Settings and make sure that the "Open Empty Editor On Start" button under Core Settings is selected.
4. Add a project using File -> Add Project Folder and open a file from that project.
5. Open another file in a different window by right-clicking on it and selecting "Open in New Window".
6. Without closing any tabs or removing any projects, quit the program. This should be done by right-clicking the Atom icon in the menu bar and selecting "Quit". Make sure "Open Empty Editor On Start" is still selected.
7. Open Atom again.
8. All the tabs from before are now open again.

**Comment by gardwired** [comment]
debian 8 atom 1.6

first off, thanks to atom team.  this is a really nice editor and am replacing sublime 3 with it.  

it appears to me that there is no multi-window support for remember session.  i never check "open empty editor" as i always want my window to remember state.   one window works fine.  if i open one and open a few files and panes and then open a new blank window and close the blank one LAST, it loses state of the first one with the files when i reopen.  i get a blank screen.  if i close the one with the open files last and then reopen, i get my correct window.  

so: LAST window closed sets the state.  

+ 1 for multi-window support for sessions.

**Comment by leobasilio** [comment]
Hi. The issue you are reporting is actually unrelated to the setting discussed here. It is due to the way state saving was implemented. Whenever a window is closed, its state is cleared, except when the application is quitting. That's why only the last window's state persists. The user usually quits by closing window by window until none is left. Saving the state for multiple windows would require the user to quit the application as a whole, and I don't think it's currently possible.

**Comment by gardwired** [comment]
gotcha.  it seems that if i just use the file > quit option i get all my windows back on restart so i just have to remember to do that before a reboot.  thanks...gg

**Comment by delliott2016** [comment]
leo, has your bug fix been accepted?

**Comment by leobasilio** [comment]
@delliott2016 Not yet, and I don't feel like it's going to be any time soon... =(

**Comment by delliott2016** [comment]
Hmm, have you any idea why? I've tested it and it works. Is it because of side effects or something else that the fix causes?

I should note that with this fix I've experienced Atom loading with a huge number of windows all at once. Instead of just one blank one I end up with 8 or 9 sometimes. Not sure if it's related, but I never saw it happen before.

Hope to see this fixed soon.

**Comment by leobasilio** [comment]
I think there are too many pull requests to be reviewed and this one might be considered a minor issue. There are PRs older than mine that have not been reviewed yet. 

I don't know whether this multiple windows thing is related to the fix. There have been many new commis since then and I haven't been following them to know if some could have broken the fix.

**Comment by delliott2016** [comment]
The problem I noted is unrelated to your bug fix, thankfully. All is well.

Most likely the issue. Guess it's just back to waiting then. Good job on coming up with a solution!

**Comment by gardwired** [comment]
just wanted to say that atom 1.7.1 fixed my issue:  multiple windows remembered after clean reboot.  no longer have to remember to explicitly quit atom.  thanks!

**Comment by delliott2016** [comment]
Should this be marked closed then by the update? Has anyone tested this on the new version?

**Comment by orschiro** [comment]
@delliott2016,

Just tested version 1.7.2 on Ubuntu 16.04 and it does work!

**Comment by sp82** [comment]
Atom 1.7.2 on Windows Server 2008 still doesn't work, I made a fresh installation a week ago.

**Comment by Kingdaro** [comment]
It still doesn't work for me either, Windows 10.

**Comment by leobasilio** [comment]
It doesn't work. Atom 1.7.2 + Windows 8.1.

**Comment by delliott2016** [comment]
Ubuntu 14.04 LTS does not work, even on Atom 1.9.0-dev-877601d.

Mac OS X El Capitan v10.11.4 does not work either, same version of Atom. Although I should note that the problem, as reported earlier on this thread, is the exact opposite of the problem on Windows and Linux.

Looks like we still need to wait on the PR, then.

**Comment by orschiro** [comment]
> @delliott2016,
Just tested version 1.7.2 on Ubuntu 16.04 and it does work!

Dear all, 

I'm correcting my above comment. It **doesn't** work on Ubuntu 16.04. Not sure what I did differently the first time that I was presented with a blank window but after a few more testings, I found that it still shows the previously opened ones.

**Comment by bigUNO** [comment]
Is there any way to get a completely blank (no projects, no tabs opened) atom instance every time I launch the app? "Open Empty Editor On Start" doesn't work.

**Comment by delliott2016** [comment]
@bigUNO I believe that's what the "Open Empty Editor on Start" is supposed to do. Until it's fixed I would say just close out of everything yourself.

**Comment by Persing** [comment]
In my download of the latest version of atom for windows and my build from the latest master branch source, "Open Empty Editor On Start" has no effect. From what I am able to tell the function of this feature is to open a new empty text buffer on startup if there are no other buffers open at the time of launching atom. I made a pull request with this feature fixed and I also added in another menu option that will make atom open empty on every launch, because it seems that is what many people wanted "Open Empty Editor On Start" to do when enabled. Here is a link to the pull request and a screen capture of the results, https://github.com/atom/atom/pull/11688

**Comment by gnomed** [comment]
Seeing this on latest version and the latest beta. Ubuntu 15.10. I have been waiting months for this feature and it doesn't freaking work -_-.

**Comment by ExClouds** [comment]
Still doesn't work..  Ubuntu 16 &  windows 10 64bit & windows 7 64bit..

**Comment by Paviluf** [comment]
Can we have this feature working please ?
It seems really easy to implement it.

**Comment by leobasilio** [comment]
The pull request is already there. Now it's up to Atom's team.

**Comment by delliott2016** [comment]
Believe it or not, this issue goes beyond just opening empty when you want it to. On Mac (as described in the long past way farther up), it _always_ opens empty, which makes the program really hard to use. I think maybe someone should bring that up with the developers if we want to get this through. (Although I technically haven't tested to see if this fix solves that on the Mac. That might be worth doing soon.)

Updating the PR with some extra information helped one of mine get noticed and accepted. @leobasilio If you want to try saying something on it, maybe it'll help them notice it more?

**Comment by Persing** [comment]
I think that the problem on Mac might be completely separate issue, because the open empty editor function is only called when the check box is checked. Then when it's called it only opens a blank text editor if there are no other buffers open. The function never actually closes anything. 

As for this bug, there are at least two PR that fix this problem that I am aware of. It's up to Atom's developers team to review and accept one.

**Comment by delliott2016** [comment]
@Persing true that it might be a different issue. 

About your solution, though, I like the approach in closing things when you first open the application. But I have a few issues with it:

- I think your definition of "Open Empty Editor on Start", although correct within the context of your fix, is not what most people will expect. Users (at least the ones who post here), expect the all of Atom to open empty, no tabs and no projects. To them, the untitled tab is just an extra nicety, not the main intention of clicking the button.
- If we go with your definition, then "Open Empty Editor on Start" becomes useless. What does it do? Open an untitled tab if there are no other tabs open on startup? That doesn't even make sense as a feature, because all the user has to do is open Atom with your feature unchecked, then open a new tab. The current feature would be better removed.
- Your implementation asks users to close tabs when they open Atom, tabs they can't even see (because they haven't been fully loaded). Users would not like having to choose whether or not to save old work when they first open the application. After all, they may not even remember it. In that case, I believe users would expect all work from before to close automatically without saving. Remember that they would not have unchecked your feature unless they wanted it all closed.
- The wording feels backward. I would personally prefer the feature to read "Close All Tabs on Start" and have to check it to make it happen.

As a result, I suggest removing the "Open Empty Editor on Start" feature, changing the code to automatically close all work that was open, and use a reversed wording. Otherwise @leobasilio's solution will need to be used, which more closely follows the expected behavior.

**Comment by delliott2016** [comment]
@leobasilio If you're not too tired of thinking about this bug, I thought of something the developers might bring up when they look at it (if they haven't already). The wording of the feature is "Open Empty Editor on Start", yet your fix prevents data from being saved on close. Just to stay more true to the wording of the feature, it might be worth updating the logic of the code to remove that data on load instead.

Just a thought.

**Comment by leobasilio** [comment]
My PR prevents data from being both saved and loaded. I had to prevent saving also because some data was already loaded by the time the settings were loaded. The original code restores part of the previous state first and only after that loads the user's settings. So the setting's value (true or false) isn't available on the first step to check whether or not to restore the state in the first place. Changing that behavior would basically break all the already existing initialization logic.

I don't think think that the way the code works behind the scenes is relevant in this case. From the user's perspective, the achieved effect is all that matters. In that sense, the PR makes this option work as expected.

**Comment by Persing** [comment]
@delliott2016, I completely agree that the user expectation is very different from what this feature is actual attempting to do. In addition to the feature currently being broken, there is a large problem with the wording. I tried to change as little as possible with my fix, however seeing that it is still confusing I will clean that up. Also I agree that the new menu option is confusing too. I will clean this up too. In regards to that I might have an of a cleaner way to implement this feature that should fix the rather large problem you mention where the user can't even tell what they are saving, but it will take me a bit to work it out. I'll update the PR when that is done. There is one current work around (not that it is acceptable for release) is to hit cancel instead of save or don't save it will open that tab.
I don't think that removing the current intended feature is a good idea, because however pointless it may seem I like the idea of giving the user the option to choose how the program behaves. 

I would also like to take a moment to thank you for your detailed, clear and polite critiques.  It is very helpful.

**Comment by itsmelion** [comment]
Confirmed on MacOS X_.11.5 ElCapitan_. My session always come Blank no matter what.

**Comment by Kingdaro** [comment]
Version 1.9.1, Windows 10. Still opening previous files with the option checked.

**Comment by dav-ell** [comment]
@leobasilio any word on the PR?

**Comment by leobasilio** [comment]
@dav-ell Not a single word.

**Comment by dav-ell** [comment]
Hmm, I wonder if maybe this issue is a sign of a deeper problem with the way they handle memory. During testing of this a while back I came across some of the strangest bugs, not all which can be explained by the fixes we've seen so far. It may require more effort to truly fix this than we think.

@leobasilio It's also likely that your PR just got looked over a long time ago, and they just never went back to it. A recommendation to anyone who is interested... fix it again and submit another PR. Maybe they'll notice this time. I'll also send a message to one of the devs, who might take another look at it.

**Comment by 50Wliu** [comment]
> A recommendation to anyone who is interested... fix it again and submit another PR.

Please don't do this unless the original author is inactive or unable to finish the existing PR.  That just makes the backlog even longer.

**Comment by dav-ell** [comment]
@50Wliu Fair enough. Though the idea was to fix it differently (and better), in case the existing PRs have not been thoroughly reviewed because they were obviously not good enough solutions. I've seen a few instances where there are multiple PRs, and only one of them ends up getting reviewed and accepted, without a comment on any of the others, regardless of order. Thought that might have been the case here.

Will remember this for future issues. Thanks.

**Comment by KingPrawn22** [comment]
Such a beautiful looking piece of software that I was looking forward to make use of. Unfortunately, this bug/feature is irritating enough for me to fallback to NP++. Great shame. Really don't understand why the behaviour we are all looking for is not the default.

**Comment by takeontom** [comment]
If there's other people like me who are arriving here looking for some way to get Atom to load up blank by default...

Just add `restorePreviousWindowsOnStart: false` into the `core` section of your config.cson file

Which is an option added in Atom 1.8: http://blog.atom.io/2016/06/06/atom-1-8-and-1-9-beta.html#introduce-corerestorepreviouswindowsonstart

**Comment by KingPrawn22** [comment]
Marvellous! Works a treat. Thank you! Being a total n00b, I was having trouble finding the `config.cson` file, but then realised that since I hadn't looked at customising Atom, that the file won't exist yet. The file can be generated by going to `File > Config...`

It should then generate a small config file that represents the current setup, to which you add the above somewhere within the `core` section.

Thanks again!

**Comment by renanwille** [comment]
On my machine atom(1.9.9) even when I set `restorePreviousWindowsOnStart: false` into the core section it continues to reopen some tabs.

**Comment by PragmaticEd** [comment]
Why is this `closed`? None of these options actually open blank editor..

**Comment by lee-dohm** [comment]
<img width="473" alt="screen shot 2016-12-31 at 10 40 17 am" src="https://cloud.githubusercontent.com/assets/1038121/21578843/8dfa269a-cf45-11e6-9823-369ed0da007e.png">

**Comment by KingPrawn22** [comment]
@PragmaticEd you are right that it doesn't not work, so not sure why it was closed. The 'fixes' worked for maybe a couple of times, then again started showing previously opened files. I uninstalled it anyways and went with different software.

**Comment by nordboerg** [comment]
Ubuntu 16.04
Atom 1.13

I always get an empty editor when I launch Atom, even if the option is checked out. I'd like to see files that I left open.

**Comment by Paviluf** [comment]
Is the fix really included in 1.13, it don't seem so ?

**Comment by 50Wliu** [comment]
As @takeontom mentioned, the "Restore Previous Windows on Start" option allows you to disable restoring your last project.  It's been around since 1.8, but was added to the Settings View in 1.13 along with a modified description of the "Open Empty Editor" setting to reduce confusion.

**Comment by nordboerg** [comment]
That must be true, but it's really annoying in such an editor like Atom that it doesn't bring back my work. It worked a few versions before, but can't remember the exact version number. Should I open a new issue?

**Comment by 50Wliu** [comment]
Sorry, I was replying to the other people who were confused about why the issue was closed.
@nordboerg yes, please open a new issue and we'll look into it.

**Comment by Paviluf** [comment]
It seem that the fix will be in Atom 1.14. I just tried the beta and that worked !
I had to enabled "Open Empty Editor On Startup" and disabled "Restore Previous Windows on Start"

**Comment by eugensunic** [comment]
I'm using windows 10 and this fixed my problem:
Make sure that you have the following line in the **config.cson** file:
  _welcome:
    showOnStartup: false_
You can write it or you can just un-click the checkbox at the startup.
After that **RESTART** your computer. Closing Atom and then opening again wasn't enough for me to get it work.

**Comment by oqtvs** [comment]
I always open atom, clicking on a file and i couldn't find a set of configurations until now(1.18.0) that do not load the previous open files from that directory. My solution to use atom was to add this on the init.coffee script (found this idea on the internet):
```
process.nextTick ->
  atom.workspace.getPaneItems().forEach ->
    atom.workspace.destroyActivePaneItem()
```
If someone knows how to solve this without adding this script to init and can point me out to the right direction I would be grateful!

**Comment by maxbernyk-seek** [comment]
For me Atom is not an IDE, but a light and flexible text editor, combined with an interactive shell (ipython/hydrogen).
The biggest difference and advantage of an editor is that a user can quickly open a file for editing without dealing with heavy projects. Having restored full workspace of the folder/directory when opening a single file only is not what expected from an editor. Would be nice to have an option for that.

End up using:
```
process.nextTick ->
  atom.workspace.getPaneItems().forEach ->
    atom.workspace.destroyActivePaneItem()
```
in init.coffee, as described above.

**Comment by dannycolin** [comment]
> @maxbernyk-seek For me Atom is not an IDE, but a light [...] text editor.

Light?... Atom isn't light at all! I'm not saying it's a bad think. I'm only saying that if you want something light you should use something else than Atom. Use the right tool for the right job. 

If you are a Linux user you should definitely use Gedit (GUI) or Nano (CLI). They are light and easy to use.

**Comment by miguelroboso** [comment]
what's the status on this?
every time I open a file I keep seeing some random previously opened files as tabs. I checked the "open empty editor" in the preferences, and `restorePreviousWindowsOnStart:` is  `false` in `core`

**Comment by Ben3eeE** [comment]
@miguelroboso The status on this is that this issue was fixed a long time ago. If you have encountered something that you think is a bug in Atom then please open a new issue and fill out the issue template.

**Comment by wotanii** [comment]
@miguelroboso it's `restorePreviousWindowsOnStart: "no"` and not `false`. I guess they changed the value in the mean time.

**Comment by tim1mw** [comment]
This issue is most definitely not fixed in 1.23.3 and should be re-opened. I have the option ticked, but files open in the previous session are always re-opened regardless.

**Comment by Ben3eeE** [comment]
The open empty editor on start option has no effect on restoring window state. It's for changing if an untitled editor is automatically opened in new Atom windows or not. The `Restore Previous Windows On Start` option changes how atom restores the previous session:

![image](https://user-images.githubusercontent.com/1058982/35567659-14bcf4ac-05c6-11e8-9005-11e1f2c1dcaa.png)

**Comment by PragmaticEd** [comment]
This issue can be closed, as @Ben3eeE posted option has been added to the latest version of Atom. :) 👍
